src/rest_vol.c

```diff
@@ -2740,6 +2749,163 @@ RV_copy_object_loc_info_callback(char *HTTP_response, void *callback_data_in, vo
     return ret_value;
 } /* end RV_copy_object_loc_info_callback() */
 
+/*-------------------------------------------------------------------------
+ * Function:    RV_copy_link_name_by_index
+ *
+ * Purpose:     This callback is used to copy the name of an link
+ *              in the server's response by index. It allocates heap memory for the name,
+ *              and returns a pointer to the memory by callback_data_out.
+ *
+ * Return:      Non-negative on success/Negative on failure
+ *
+ * Programmer:  Matthew Larson
+ *              May, 2023
+ */
+herr_t
+RV_copy_link_name_by_index(char *HTTP_response, void *callback_data_in, void *callback_data_out)
+{
+    yajl_val           parse_tree = NULL, key_obj = NULL, link_obj = NULL;
+    const char        *parsed_link_name = NULL;
+    H5VL_loc_by_idx_t *idx_params       = (H5VL_loc_by_idx_t *)callback_data_in;
+    hsize_t            index            = idx_params->n;
+    char             **link_name        = (char **)callback_data_out;
+    const char        *curr_key         = NULL;
+    herr_t             ret_value        = SUCCEED;
+
+    if (!HTTP_response)
+        FUNC_GOTO_ERROR(H5E_ARGS, H5E_BADVALUE, FAIL, "HTTP response buffer was NULL");
+
+    if (NULL == (parse_tree = yajl_tree_parse(HTTP_response, NULL, 0)))
+        FUNC_GOTO_ERROR(H5E_OBJECT, H5E_PARSEERROR, FAIL, "parsing JSON failed");
+
+    if (NULL == (key_obj = yajl_tree_get(parse_tree, links_keys, yajl_t_array)))
+        FUNC_GOTO_ERROR(H5E_OBJECT, H5E_PARSEERROR, FAIL, "failed to parse links");
+
+    if (key_obj->u.array.len == 0)
+        FUNC_GOTO_ERROR(H5E_OBJECT, H5E_PARSEERROR, FAIL, "parsed link array was empty");
+
+    if (index >= key_obj->u.array.len)
+        FUNC_GOTO_ERROR(H5E_OBJECT, H5E_PARSEERROR, FAIL, "requested link index was out of bounds");
+
+    switch (idx_params->order) {
+        case (H5_ITER_DEC):
+            if (NULL == (link_obj = key_obj->u.array.values[key_obj->u.object.len - 1 - index]))
+                FUNC_GOTO_ERROR(H5E_OBJECT, H5E_PARSEERROR, FAIL, "selected link was NULL");
+            break;
+
+        case (H5_ITER_NATIVE):
+        case (H5_ITER_INC):
+        case (H5_ITER_N):
+        case (H5_ITER_UNKNOWN):
```

Comment:
cases `H5_ITER_N`, `H5_ITER_UNKNOWN` and default should return an error. Only cases `H5_ITER_NATIVE` and `H5_ITER_INC` should have the below implementation.